latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/TemplateProcessor.java

```diff
@@ -171,24 +173,25 @@ private String getTemplateName(Block block)
         return templateName;
     }
 
-    private Optional<String> renderUIXPs(String templateName, String suffix)
+    private Optional<String> renderUIXs(String templateName, String suffix)
     {
         // The XDOM template case is particular, the after UIXP needs to be located before "\end{document}" which closes
         // the latex document. To do so, the UIXP of XDOM are directly integrated in the template in Velocity.
         Optional<String> result;
         if (!templateName.equals("XDOM")) {
             List<UIExtension> extensions =
-                this.uiExtensionSupplier.getExtensions(
-                    String.format("org.xwiki.contrib.latex.%s.%s", templateName, suffix));
+                this.uiExtensionManager.getExtensions(
+                    String.format("org.xwiki.contrib.latex.%s.%s", templateName, suffix),
+                    singletonMap("sortByParameter", "order"));
             if (extensions.isEmpty()) {
                 result = Optional.empty();
             } else {
                 DefaultWikiPrinter printer = new DefaultWikiPrinter();
                 // Sort the extensions by their "order" parameter, the values with the highest order value first.
                 // Missing or invalid order values are considered to be 0.
```

Comment:
Maybe remove this comment or move it above?

Reply:
Good catch. Thanks.